.github/workflows/tests.yml

```diff
@@ -39,34 +39,48 @@ jobs:
     - name: Validate composer.json and composer.lock
       run: composer validate
 
-    #downgrade to supported php unit and remove incompatible xdebug file if it exists
+    #remove incompatible xdebug file if it exists
     - if: ${{ matrix.php-versions == '5.5' }}
-      name: PHP 5.5 specific setup
+      name: PHP 5.5
       run: |
         sudo rm -f /etc/php5/cli/conf.d/20-xdebug.ini
-        composer require --dev --ignore-platform-reqs phpunit/phpunit "4.8.36"
-
-    #downgrade to supported php unit
-    - if: ${{ matrix.php-versions <= '7.1'  && matrix.php-versions > 5.5}}
-      name: PHP 5.6-7.1 specific setup
-      run: composer require --dev --ignore-platform-reqs phpunit/phpunit "^5.7.11"
 
     #get dependencies
     - name: Install dependencies
       run: composer update ${{ matrix.composer-options }} --no-interaction --prefer-source
 
+    #php 8.x requirements
+    - if: ${{ matrix.php-versions >= '8.0' && matrix.composer-options != '' }}
+      name: PHP 8.x
+      run: composer require --dev phpunit/phpunit "^9.5" --no-interaction --prefer-source --with-all-dependencies
+
+    #php 8.1 requirements
+    - if: ${{ matrix.php-versions >= '8.1' && matrix.composer-options != '' }}
+      name: PHP 8.1
+      run: composer require --dev guzzlehttp/guzzle "^7.3" --no-interaction --prefer-source --with-all-dependencies
+
     #run tests
     - name: Run test suite
       run: make test
 
     #static analysis
-    - if: ${{ matrix.php-versions >= '7.1' && matrix.php-versions <= 7.4 && matrix.composer-options == '' }}
+    - if: ${{ matrix.php-versions >= '7.1' && matrix.php-versions < '8.0' && matrix.composer-options == '' }}
       name: Static analysis
       run: |
         composer require --dev nette/neon "^3.0"
         composer require --dev phpstan/phpstan "0.12.45"
         vendor/bin/phpstan analyse src
 
+    #static compatibility-check for all versions
+    #Only needs to be run on 1 php-version to check for all versions
```

Comment:
If there is 100% code coverage, this will be redundant, yes.  

When I added the tool, only false positives were detected, so in that regard this tool does not add much value to the project as it stands.

However, I think the output from the tool is still useful in case some new code is using non-bc-compatible syntax or function-arguments.  I think this output might be more clear than a failing unittest, where you as a developer need to be aware what exactly is not valid in all versions of php.

I've used this tool in other projects, which did have non-compatible code.  There it helped to figure out which code needed some refactoring.

If you decide the tool does not bring much value, you can simply drop 7c51a9ea79fd6dc435dd9e2b0d1e94b0743ba092 from the PR as it was added in a seperate commit.  Worst case my effort lead to a validation that all code is valid in php 5.5-8.1 :slightly_smiling_face:.